File: app/src/main/java/com/drdisagree/iconify/ui/activities/Notification.java
package com.drdisagree.iconify.ui.activities;

import android.annotation.SuppressLint;
import android.os.Bundle;

import androidx.recyclerview.widget.ConcatAdapter;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.drdisagree.iconify.R;
import com.drdisagree.iconify.ui.adapters.MenuAdapter;
import com.drdisagree.iconify.ui.adapters.NotificationAdapter;
import com.drdisagree.iconify.ui.adapters.ViewAdapter;
import com.drdisagree.iconify.ui.models.MenuModel;
import com.drdisagree.iconify.ui.models.NotificationModel;
import com.drdisagree.iconify.ui.utils.ViewBindingHelpers;
import com.drdisagree.iconify.ui.views.LoadingDialog;

import java.util.ArrayList;

public class Notification extends BaseActivity {

    LoadingDialog loadingDialog;

    @SuppressLint("SetTextI18n")
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_notifications);

        // Header
        ViewBindingHelpers.setHeader(this, findViewById(R.id.collapsing_toolbar), findViewById(R.id.toolbar), R.string.activity_title_notification);

        // Loading dialog while enabling or disabling pack
        loadingDialog = new LoadingDialog(this);

        // RecyclerView
        RecyclerView container = findViewById(R.id.notifications_container);
        container.setLayoutManager(new LinearLayoutManager(this));
        ConcatAdapter adapter = new ConcatAdapter(initActivityItems(), new ViewAdapter(this, R.layout.view_divider), initNotifItems());
        container.setAdapter(adapter);
        container.setHasFixedSize(true);
    }

    private MenuAdapter initActivityItems() {
        ArrayList<MenuModel> notif_activity_list = new ArrayList<>();

        notif_activity_list.add(new MenuModel(NotificationPixel.class, getResources().getString(R.string.activity_title_pixel_variant), getResources().getString(R.string.activity_desc_pixel_variant), R.drawable.ic_pixel_device));

        return new MenuAdapter(this, notif_activity_list);
    }

    private NotificationAdapter initNotifItems() {
        ArrayList<NotificationModel> notif_list = new ArrayList<>();

        notif_list.add(new NotificationModel("Default", R.drawable.notif_default));
        notif_list.add(new NotificationModel("Layers", R.drawable.notif_layers));
        notif_list.add(new NotificationModel("Thin Outline", R.drawable.notif_thin_outline));
        notif_list.add(new NotificationModel("Bottom Outline", R.drawable.notif_bottom_outline));
        notif_list.add(new NotificationModel("Neumorph", R.drawable.notif_neumorph));
        notif_list.add(new NotificationModel("Stack", R.drawable.notif_stack));
        notif_list.add(new NotificationModel("Side Stack", R.drawable.notif_side_stack));
        notif_list.add(new NotificationModel("Outline", R.drawable.notif_outline));
        notif_list.add(new NotificationModel("Leafy Outline", R.drawable.notif_leafy_outline));
        notif_list.add(new NotificationModel("Lighty", R.drawable.notif_lighty));
        notif_list.add(new NotificationModel("Neumorph Outline", R.drawable.notif_neumorph_outline));
        notif_list.add(new NotificationModel("Cyberponk", R.drawable.notif_cyberponk));
        notif_list.add(new NotificationModel("Cyberponk v2", R.drawable.notif_cyberponk_v2));
        notif_list.add(new NotificationModel("Thread Line", R.drawable.notif_thread_line));
        notif_list.add(new NotificationModel("Faded", R.drawable.notif_faded));
        notif_list.add(new NotificationModel("Dumbbell", R.drawable.notif_dumbbell));
        notif_list.add(new NotificationModel("Semi Transparent", R.drawable.notif_semi_transparent));
        notif_list.add(new NotificationModel("Pitch Black", R.drawable.notif_pitch_black));
        notif_list.add(new NotificationModel("Duoline", R.drawable.notif_duoline));
        notif_list.add(new NotificationModel("iOS", R.drawable.notif_ios));

        return new NotificationAdapter(this, notif_list, loadingDialog, "NFN");
    }

    @Override
    public void onDestroy() {
        loadingDialog.hide();
        super.onDestroy();
    }
}

File: app/src/main/java/com/drdisagree/iconify/ui/fragments/home/Notification.kt
package com.drdisagree.iconify.ui.fragments.home

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.ConcatAdapter
import androidx.recyclerview.widget.LinearLayoutManager
import com.drdisagree.iconify.R
import com.drdisagree.iconify.databinding.FragmentNotificationBinding
import com.drdisagree.iconify.ui.adapters.MenuAdapter
import com.drdisagree.iconify.ui.adapters.NotificationAdapter
import com.drdisagree.iconify.ui.adapters.SectionTitleAdapter
import com.drdisagree.iconify.ui.base.BaseFragment
import com.drdisagree.iconify.ui.dialogs.LoadingDialog
import com.drdisagree.iconify.ui.models.MenuModel
import com.drdisagree.iconify.ui.models.NotificationModel
import com.drdisagree.iconify.ui.utils.ViewHelper.setHeader

class Notification : BaseFragment() {

    private lateinit var binding: FragmentNotificationBinding
    private var loadingDialog: LoadingDialog? = null

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        binding = FragmentNotificationBinding.inflate(inflater, container, false)
        val view: View = binding.getRoot()

        // Header
        setHeader(
            requireContext(),
            getParentFragmentManager(),
            binding.header.toolbar,
            R.string.activity_title_notification
        )

        // Loading dialog while enabling or disabling pack
        loadingDialog = LoadingDialog(requireContext())

        // RecyclerView
        binding.notificationsContainer.setLayoutManager(LinearLayoutManager(requireContext()))

        val adapter = ConcatAdapter(
            initActivityItems(),
            SectionTitleAdapter(
                requireContext(),
                R.layout.view_section_title_notif,
                R.string.notification_styles
            ),
            initNotificationItems()
        )

        binding.notificationsContainer.setAdapter(adapter)
        binding.notificationsContainer.setHasFixedSize(true)

        return view
    }

    private fun initActivityItems(): MenuAdapter {
        val notificationActivityList = ArrayList<MenuModel>().apply {
            add(
                MenuModel(
                    NotificationPixel(),
                    resources.getString(R.string.activity_title_pixel_variant),
                    resources.getString(R.string.activity_desc_pixel_variant),
                    R.drawable.ic_pixel_device
                )
            )
        }

        return MenuAdapter(
            parentFragmentManager,
            requireContext(),
            notificationActivityList
        )
    }

    private fun initNotificationItems(): NotificationAdapter {
        val notificationList = ArrayList<NotificationModel>().apply {
            add(
                NotificationModel(
                    "Default",
                    R.drawable.notif_default
                )
            )
            add(
                NotificationModel(
                    "Layers",
                    R.drawable.notif_layers
                )
            )
            add(
                NotificationModel(
                    "Thin Outline",
                    R.drawable.notif_thin_outline
                )
            )
            add(
                NotificationModel(
                    "Bottom Outline",
                    R.drawable.notif_bottom_outline
                )
            )
            add(
                NotificationModel(
                    "Neumorph",
                    R.drawable.notif_neumorph
                )
            )
            add(
                NotificationModel(
                    "Stack",
                    R.drawable.notif_stack
                )
            )
            add(
                NotificationModel(
                    "Side Stack",
                    R.drawable.notif_side_stack
                )
            )
            add(
                NotificationModel(
                    "Outline",
                    R.drawable.notif_outline
                )
            )
            add(
                NotificationModel(
                    "Leafy Outline",
                    R.drawable.notif_leafy_outline
                )
            )
            add(
                NotificationModel(
                    "Lighty",
                    R.drawable.notif_lighty
                )
            )
            add(
                NotificationModel(
                    "Neumorph Outline",
                    R.drawable.notif_neumorph_outline
                )
            )
            add(
                NotificationModel(
                    "Cyberponk",
                    R.drawable.notif_cyberponk
                )
            )
            add(
                NotificationModel(
                    "Cyberponk v2",
                    R.drawable.notif_cyberponk_v2
                )
            )
            add(
                NotificationModel(
                    "Thread Line",
                    R.drawable.notif_thread_line
                )
            )
            add(
                NotificationModel(
                    "Faded",
                    R.drawable.notif_faded
                )
            )
            add(
                NotificationModel(
                    "Dumbbell",
                    R.drawable.notif_dumbbell
                )
            )
            add(
                NotificationModel(
                    "Semi Transparent",
                    R.drawable.notif_semi_transparent
                )
            )
            add(
                NotificationModel(
                    "Pitch Black",
                    R.drawable.notif_pitch_black
                )
            )
            add(
                NotificationModel(
                    "Duoline",
                    R.drawable.notif_duoline
                )
            )
            add(
                NotificationModel(
                    "iOS",
                    R.drawable.notif_ios
                )
            )
        }

        return NotificationAdapter(
            requireContext(),
            notificationList,
            loadingDialog!!,
            "NFN"
        )
    }

    override fun onDestroy() {
        loadingDialog?.dismiss()

        super.onDestroy()
    }
}